DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -2614,27 +2619,39 @@ public partial class DMParser : Parser<Token> {
                         if (closed) break;
                     }
 
-                    Consume(new TokenType[] { TokenType.DM_Identifier, TokenType.DM_Null }, "Expected value type");
-                    switch (typeToken.Text) {
-                        case "anything": type |= DMValueType.Anything; break;
-                        case "null": type |= DMValueType.Null; break;
-                        case "text": type |= DMValueType.Text; break;
-                        case "obj": type |= DMValueType.Obj; break;
-                        case "mob": type |= DMValueType.Mob; break;
-                        case "turf": type |= DMValueType.Turf; break;
-                        case "num": type |= DMValueType.Num; break;
-                        case "message": type |= DMValueType.Message; break;
-                        case "area": type |= DMValueType.Area; break;
-                        case "color": type |= DMValueType.Color; break;
-                        case "file": type |= DMValueType.File; break;
-                        case "command_text": type |= DMValueType.CommandText; break;
-                        case "sound": type |= DMValueType.Sound; break;
-                        case "icon": type |= DMValueType.Icon; break;
-                        case "opendream_unimplemented": type |= DMValueType.Unimplemented; break;
-                        case "opendream_compiletimereadonly": type |= DMValueType.CompiletimeReadonly; break;
-                        default: Error("Invalid value type '" + typeToken.Text + "'"); break;
-                    }
 
+                    if (!Check(new TokenType[] { TokenType.DM_Identifier, TokenType.DM_Null })) {
+                        // Proc return types
+                        if (allowPath) {
```

Comment:
It would be better for error recovery if this always tried to parse a path. You can accept or deny the path based on `allowPath` afterwards.

Right now `var/a = input("A") as /list` gives multiple errors:
```
Error OD0001 at code.dm:183:30: Expected value type
Error OD0000 at code.dm:184:2: Unknown identifier "list"
```